doc/tor.md

```diff
@@ -216,10 +216,30 @@ for normal IPv4/IPv6 communication, use:
 
     ./bitcoind -onion=127.0.0.1:9050 -externalip=7zvj7a2imdgkdbg4f2dryd5rgtrn7upivr5eeij4cicjh65pooxeshid.onion -discover
 
-## 4. Privacy recommendations
+## 4. Security recommendations
 
-- Do not add anything but Bitcoin Core ports to the onion service created in section 3.
+Do not add anything but Bitcoin Core ports to the onion service created in section 3.
   If you run a web service too, create a new onion service for that.
   Otherwise it is trivial to link them, which may reduce privacy. Onion
   services created automatically (as in section 2) always have only one port
   open.
+  
+**Sybil Attacks**
+
+  If you only make random Tor connections, you're much more vulnerable to Sybil attacks.
+  As Tor addresses may be created at no cost, an attacker can potentially flood the network with many Tor
+  nodes and receive all of the outbound Tor connections an `onlynet=onion` node makes.
+
+  This is significantly less a concern if you make `-addnode` connections to trusted peers
+  (even if they're onion addresses). It's also alleviated with IPv4/IPv6 connections (especially when using the `-asmap`
+  configuration option) due to the cost of obtaining IPs in many networks.
+
+  If all of your connections are controlled by a Sybil attacker, they can easily prevent you from seeing confirmed
+  transactions and, with more difficulty, even trick your node into falsely reporting a transaction as confirmed
+  on the blockchain with the most cumulative “chainwork”.
+
+**Network Partitioning**
+
+  If too many nodes use `onlynet=onion`, it could become difficult for onion nodes to communicate with clearnet nodes,
+  preventing the Tor network from seeing recent transactions and blocks. It is essential that some nodes access both
+  clearnet and Tor.
```

Comment:
I am not sure I grasp this. Do you mean the following:

If all nodes that connect to the Tor network use `onlynet=onion`, it would be impossible for onion nodes to communicate with the rest of the network...

Reply:
Those nodes could still have incoming connections.
But the concern is the network partitioning risk from having a super majority of Tor nodes only having outgoing connections to the Tor network.
Many people think Tor-only is the safest, but there are many risks which I think is important to state in the docs

Reply:
"it could become difficult for onion nodes to communicate with clearnet nodes" -> maybe "clearnet, I2P, and CJDNS nodes"

Reply:
> If too many nodes use `onlynet=onion`

This is probably not what you want to say. It means that if now many (e.g. 1000) new nodes are added that use `onlynet=onion` this would be a problem somehow. I think the concern here is about the ratio of "nodes that access the Tor network + other networks" / "nodes that access the Tor network only".

> preventing the Tor network from seeing recent transactions and blocks

This implies that recent transactions and blocks are created outside of the Tor network. But some could be first/only broadcast in the Tor network.

Suggestion:

If too many of the Tor nodes use `onlynet=onion`, it could become difficult for Tor nodes to communicate with the rest of the network, increasing the chance of network partitioning. It is essential that some nodes access Tor nodes and nodes on other networks.